### PR TITLE
[WIP]Feature/update script http 032

### PR DESCRIPTION
**Pull request by istoimenova**

### CODE REVIEW

**Comment by istoimenova** [comment]
@dboltovskyi, please review.

File: test_scripts/Polices/build_options/032_ATF_P_TC_PTU_Retry_Sequence_Retry_Timeout_Expiration_HTTP.lua [hunk 1]
```diff
@@ -103,14 +100,15 @@ function Test:TestStep_Retry_Timeout_Expiration()
   time_wait[5] = sec_btw_ret[3] + sec_btw_ret[4] + timeout_after_x_seconds
   time_wait[6] = sec_btw_ret[4] + sec_btw_ret[5] + timeout_after_x_seconds
   total_time = (time_wait[1] + time_wait[2] + time_wait[3] + time_wait[4] + time_wait[5] + time_wait[6])*1000
+  print("Wait " .. total_time .. "msec")
```
Comment (dboltovskyi) [inline test_scripts/Polices/build_options/032_ATF_P_TC_PTU_Retry_Sequence_Retry_Timeout_Expiration_HTTP.lua]: @istoimenova, In lines 97 - 101 time_wait[x] calculated incorrectly
According to req-t formula is the following:
```
Example:

        "timeout_after_x_seconds":30,
        "seconds_between_retries":[1, 5, 25, 125, 625]

SDL->app ID: send SnapshotPT via OnSystemRequest
wait during timeout (e.g. 30s)
PTU not received
SDL->app ID: send SnapshotPT via OnSystemRequest
wait during t1 + timeout (e.g. 1s + 30s=31s)
PTU not received
SDL->app ID: send SnapshotPT via OnSystemRequest
wait during t2 + t1 + timeout (e.g. 5s + 31s + 30s=66s)
PTU not received
SDL->app ID: send SnapshotPT via OnSystemRequest
wait during t3 + t2 + timeout (e.g. 25s + 66s + 30s=121s)
etc.
```

Please use the correct one from 032_ATF_P_TC_PTU_Restarting_Policy_Table_Exchange_HTTP.lua script

**Comment by istoimenova** [comment]
[WIP] - Clarified with Igor Kovalenko

File: test_scripts/Polices/build_options/ATF_P_TC_PTU_Cycleing_Through_The_URLs_During_Retry_Sequence.lua [hunk 2]
```diff
@@ -35,7 +35,7 @@ local r_expected = {
   "http://policies.domain1.ford.com/api/policies",
   "http://policies.domain2.ford.com/api/policies",
   "http://policies.domain3.ford.com/api/policies",
-  "http://policies.domain4.ford.com/api/policies"}
+"http://policies.domain4.ford.com/api/policies"}
```
Comment (istoimenova) [inline test_scripts/Polices/build_options/ATF_P_TC_PTU_Cycleing_Through_The_URLs_During_Retry_Sequence.lua]: @dboltovskyi, script is removed.

File: test_scripts/Polices/build_options/032_ATF_P_TC_PTU_Retry_Sequence_Retry_Timeout_Expiration_HTTP.lua (hunk 1 above)
Comment (istoimenova) [inline test_scripts/Polices/build_options/032_ATF_P_TC_PTU_Retry_Sequence_Retry_Timeout_Expiration_HTTP.lua]: @dboltovskyi, done in f5a95380412269faf771e643fbe0cd08365bf7a6

File: test_scripts/Polices/build_options/032_ATF_P_TC_PTU_Retry_Sequence_Retry_Timeout_Expiration_HTTP.lua [hunk 3]
```diff
@@ -75,42 +71,45 @@ function Test:TestStep_OnStatusUpdate_UPDATE_NEEDED_new_PTU_request()
...
   local total_time
 
-  time_wait[1] = timeout_after_x_seconds
-  time_wait[2] = sec_btw_ret[1] + timeout_after_x_seconds
-  time_wait[3] = sec_btw_ret[1] + sec_btw_ret[2] + timeout_after_x_seconds
-  time_wait[4] = sec_btw_ret[2] + sec_btw_ret[3] + timeout_after_x_seconds
-  time_wait[5] = sec_btw_ret[3] + sec_btw_ret[4] + timeout_after_x_seconds
-  time_wait[6] = sec_btw_ret[4] + sec_btw_ret[5] + timeout_after_x_seconds
+  time_wait[1] = timeout_after_x_seconds -- 30
+  time_wait[2] = timeout_after_x_seconds + sec_btw_ret[1] -- 31
```
Comment (dboltovskyi) [inline test_scripts/Polices/build_options/032_ATF_P_TC_PTU_Retry_Sequence_Retry_Timeout_Expiration_HTTP.lua]: @istoimenova, Please correct:
time_wait[2] = timeout_after_x_seconds + sec_btw_ret[1] ->
time_wait[2] = timeout_after_x_seconds + sec_btw_ret[1] + time_wait[1]

Comment (istoimenova) [inline test_scripts/Polices/build_options/032_ATF_P_TC_PTU_Retry_Sequence_Retry_Timeout_Expiration_HTTP.lua]: @dboltovskyi, as Lua start counting indexes from 1, here time_wait[2] is equal to: _t1_retry = to + t1,_ 
See diagram: https://adc.luxoft.com/confluence/download/attachments/281284982/Policy_retry_sequence_Genivi.png?version=5&modificationDate=1447360904000&api=v2
Also additional information for counting times or retry sequence you can find in Policy 'second_between_retries' in retry sequence

Comment (dboltovskyi) [inline test_scripts/Polices/build_options/032_ATF_P_TC_PTU_Retry_Sequence_Retry_Timeout_Expiration_HTTP.lua]: @istoimenova, According to "Formula for retry sequence timeout" clarification implemented formula is correct.

File: test_scripts/Polices/build_options/032_ATF_P_TC_PTU_Retry_Sequence_Retry_Timeout_Expiration_HTTP.lua [hunk 4]
```diff
@@ -127,13 +126,14 @@ function Test:TestStep_Retry_Timeout_Expiration()
   if(time_wait == 0) then time_wait = 63000 end
 
   EXPECT_NOTIFICATION("OnSystemRequest", { requestType = "HTTP", fileType = "JSON"}):Timeout(total_time+60000):Times(5)
```
Comment (dboltovskyi) [inline test_scripts/Polices/build_options/032_ATF_P_TC_PTU_Retry_Sequence_Retry_Timeout_Expiration_HTTP.lua]: @istoimenova, Please change the expected value of fileType to "BINARY"

File: test_scripts/Polices/build_options/032_ATF_P_TC_PTU_Retry_Sequence_Retry_Timeout_Expiration_HTTP.lua [hunk 5]
```diff
@@ -148,4 +148,4 @@ function Test.Postcondition_Stop_SDL()
   StopSDL()
 end
 
-return Test
\ No newline at end of file
+return Test
```
Comment (dboltovskyi) [inline test_scripts/Polices/build_options/032_ATF_P_TC_PTU_Retry_Sequence_Retry_Timeout_Expiration_HTTP.lua]: @istoimenova, Please restore original sdl_preloaded_pt.json file in postconditions

File: test_scripts/Polices/build_options/032_ATF_P_TC_PTU_Retry_Sequence_Retry_Timeout_Expiration_HTTP.lua (hunk 3 above)
Comment (istoimenova) [inline test_scripts/Polices/build_options/032_ATF_P_TC_PTU_Retry_Sequence_Retry_Timeout_Expiration_HTTP.lua]: @dboltovskyi, thank you for clarification.

Comment (istoimenova) [inline test_scripts/Polices/build_options/032_ATF_P_TC_PTU_Retry_Sequence_Retry_Timeout_Expiration_HTTP.lua]: @dboltovskyi, thank you for clarification.

File: test_scripts/Polices/build_options/032_ATF_P_TC_PTU_Retry_Sequence_Retry_Timeout_Expiration_HTTP.lua (hunk 4 above)
Comment (istoimenova) [inline test_scripts/Polices/build_options/032_ATF_P_TC_PTU_Retry_Sequence_Retry_Timeout_Expiration_HTTP.lua]: @dboltovskyi, done in 3dda90a4334d9aaa093d9f4ce2fdb190c0d8b274

File: test_scripts/Polices/build_options/032_ATF_P_TC_PTU_Retry_Sequence_Retry_Timeout_Expiration_HTTP.lua (hunk 5 above)
Comment (istoimenova) [inline test_scripts/Polices/build_options/032_ATF_P_TC_PTU_Retry_Sequence_Retry_Timeout_Expiration_HTTP.lua]: @dboltovskyi, done in 3dda90a4334d9aaa093d9f4ce2fdb190c0d8b274

**Pull request merged**